solana/solana-ibc/programs/solana-ibc/src/chain.rs

```diff
@@ -150,6 +150,23 @@ impl ChainData {
             .map_err(into_error)
     }
 
+    /// Returns the validator data with stake and rewards
+    pub fn get_validator(
+        &self,
+        validator: Pubkey,
+    ) -> Result<
+        Option<blockchain::candidates::Candidate<PubKey>>,
+        ChainNotInitialised,
+    > {
+        let inner = self.get()?;
+        Ok(inner
+            .manager
+            .get_candidates()
+            .iter()
+            .find(|&c| c.pubkey == validator.into())
+            .cloned())
+    }
```

Comment:
Candidate isn’t the same as a validator. If you need to find a validator by a pubkey, you need to look at  ChainManager::next_epoch field.

Reply:
oh i see, makes sense.